tests/test-sessions/test-infinispan-sessions/src/test/java/org/eclipse/jetty/server/session/remote/RemoteInfinispanTestSupport.java

```diff
@@ -31,42 +31,93 @@
+
+    static GenericContainer infinispan;
 
     static
     {
         try
         {
-            String host = System.getProperty("hotrod.host", "127.0.0.1");
-
+            Testcontainers.exposeHostPorts(11222);
+            long start = System.currentTimeMillis();
+            String infinispanVersion = System.getProperty("infinispan.docker.image.version", "9.4.8.Final");
+            infinispan =
+                new GenericContainer(System.getProperty("infinispan.docker.image.name", "jboss/infinispan-server") +
+                                         ":" + infinispanVersion)
+                    .withEnv("APP_USER","theuser")
+                    .withEnv("APP_PASS","foobar")
+                    .withEnv("MGMT_USER", "admin")
+                    .withEnv("MGMT_PASS", "admin")
+                    .waitingFor(new LogMessageWaitStrategy()
+                                    .withRegEx(".*Infinispan Server.*started in.*\\s"))
+                    .withExposedPorts(4712,4713,8088,8089,8443,9990,9993,11211,11222,11223,11224)
+                    .withCopyFileToContainer(MountableFile.forClasspathResource("remote-session-test.xml"),
+                                              "/opt/jboss/infinispan-server/standalone/configuration/remote-session-test.xml")
+                    .withLogConsumer(new Slf4jLogConsumer(INFINISPAN_LOG));
+            // we could simply use `standalone` as well but... :)
+            infinispan =  infinispan.withCommand("-c remote-session-test.xml");
+            infinispan.start();
+            String host = infinispan.getContainerIpAddress();
+            System.setProperty("hotrod.host", host);
+            int port = infinispan.getMappedPort(11222);
+
+            LOG.info("Infinispan container started for {}:{} - {}ms", host, port,
+                     System.currentTimeMillis() - start);
             SearchMapping mapping = new SearchMapping();
             mapping.entity(SessionData.class).indexed().providedId()
                 .property("expiry", ElementType.METHOD).field();
 
             Properties properties = new Properties();
             properties.put(Environment.MODEL_MAPPING, mapping);
 
-            ConfigurationBuilder clientBuilder = new ConfigurationBuilder();
-            clientBuilder.withProperties(properties).addServer().host(host).marshaller(new ProtoStreamMarshaller());
+            ConfigurationBuilder configurationBuilder = new ConfigurationBuilder().withProperties(properties)
+                .addServer().host(host).port(port)
+                // we just want to limit connectivity to list of host:port we knows at start
+                // as infinispan create new host:port dynamically but due to how docker expose host/port we cannot do that
+                .clientIntelligence(ClientIntelligence.BASIC)
+                .marshaller(new ProtoStreamMarshaller());
+
+            if (infinispanVersion.startsWith("1"))
```

Comment:
What's infinispan version 1? Support for infinispan started at version 7.

Reply:
from infinispan 10 onward some setup has changed but I agree it's a bit complicated. Do we really want something such dynamic? or testing with a fixed version is fine?

Reply:
Well we should at least test against the version we compile against (currently 9.4.8.Final). Anything else is a nice-to-have.